sycl/source/detail/scheduler/commands.cpp

```diff
@@ -388,6 +394,40 @@ void ExecCGCommand::flushStreams() {
   }
 }
 
+cl_int UpdateHostRequirementCommand::enqueueImp() {
+  std::vector<RT::PiEvent> RawEvents;
+  RawEvents =
+      Command::prepareEvents(detail::getSyclObjImpl(MQueue->get_context()));
+  RT::PiEvent &Event = MEvent->getHandleRef();
+  Command::waitForEvents(MQueue, RawEvents, Event);
+
+  assert(MAllocaForReq && "Expected valid alloca command");
+  assert(MReqToUpdate && "Expected valid requirement");
+  MReqToUpdate->MData = MAllocaForReq->getMemAllocation();
+  return CL_SUCCESS;
+}
+
+void UpdateHostRequirementCommand::printDot(std::ostream &Stream) const {
+  Stream << "\"" << this << "\" [style=filled, fillcolor=\"#f1337f\", label=\"";
+
+  Stream << "ID = " << this << "\n";
+  Stream << "UPDATE REQ ON " << deviceToString(MQueue->get_device()) << "\\n";
+  Stream << "Is sub buffer: " << std::boolalpha
+         << (MStoredRequirement.MOffsetInBytes != 0) << "\\n";
```

Comment:
Should this check be changed to handle cases when offset is 0?

Reply:
Sure, forgot about this